cmake/CMakeLists.txt

```diff
@@ -350,6 +350,9 @@ if (onnxruntime_USE_ROCM)
   else()
     message(FATAL_ERROR "Cannot determine ROCm version string")
   endif()
+  if (ROCM_VERSION_DEV VERSION_GREATER_EQUAL "6.2")
+      set(CMAKE_HIP_ARCHITECTURES "gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx940;gfx941;gfx942;gfx1200;gfx1201")
```

Comment:
gfx940;gfx941;gfx942 need ROCM 6.1  (early version 6.0.x might also work).

I suggest to upgrade ROCm in nuget pipeline to at least 6.1.

Reply:
@tianleiwu is upgrading the ROCm version in the pipeline absolutely necessary? Peixuan created this PR to avoid upgrading the ROCm version since doing so will take too much time, and we're trying to get this pipeline to pass to cut the ORT 1.20 release branch tomorrow.

Reply:
The upgrade is in progress, and if it can be tested before 10/16, the upgrade PR will be created.